### DISCUSSION TITLE
Cruiser 3.0.42

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#470] [Announcements]
Cruiser 3.0.42 is on [Google Play](https://play.google.com/store/apps/details?id=gr.talent.cruiser):

- https://github.com/devemux86/cruiser/discussions/142
(new default routing mode)

- https://github.com/devemux86/cruiser/discussions/449
(Settings | Routing | Advanced)

- https://github.com/devemux86/cruiser/discussions/441 (internal)

- https://github.com/devemux86/cruiser/discussions/454#discussioncomment-8570130

- https://github.com/devemux86/cruiser/discussions/444

<img src="https://github.com/devemux86/cruiser/assets/3484020/5babf672-86fa-4895-a5c0-196bf9b17f01" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/9623dcba-d65d-45e7-b995-0c303f1a1d1c" width="50%"/>
<img src="https://github.com/devemux86/cruiser/assets/3484020/dab8b0c3-b063-4d43-b6ad-8305b4a85ebd" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/766d2754-d666-4158-a429-a3301c1e2296" width="50%"/>

**Comment by 1maxi2** [comment]
Gibt es die Möglichkeit von 3.0.41 auf die 3.0.40 zurück zu wechseln ?

**Reply by devemux86** [reply]
@1maxi2 

The new version is full of improvements and fixes. Only the latest version is supported by Google Play.

Die neue Version ist voll von Verbesserungen und Korrekturen. Nur die neueste Version wird von Google Play unterstützt.

**Reply by bios4** [reply]
> Gibt es die Möglichkeit von 3.0.41 auf die 3.0.40 zurück zu wechseln ?

Why do you want to switch back to 3.0.40 @1maxi2?
Maybe @devemux86 could provide the APK of 3.0.40 on the Dropbox share like the betas?

**Reply by devemux86** [reply]
There are new features waiting in line for testing, so we will have a new Beta period soon.

For now, if you find anything in version 3.0.41, please report it so we can improve / fix it. :slightly_smiling_face:

**Reply by 1maxi2** [reply]
Wenn das zurückwechseln nicht einfach möglich ist dann ist das für mich O.k.
Es ist nur so das ich mich bei der alten Version ganz gut umgehen konnte.
Die neue Version überfordert mich mit den Möglichkeiten. Ich hätte mir eine einfache  Möglichkeit gewünscht Autobahnen zuzulassen oder auch nicht so wie in kurviger 1. Aber vielleicht gibt es mal eine Anleitung.🤔

**Reply by devemux86** [reply]
@1maxi2 

The avoidance options are the same. Simply select which types of streets you want to avoid.

Die Vermeidungsoptionen sind dieselben. Wählen Sie einfach aus, welche Straßentypen Sie vermeiden möchten.

<img src="https://github.com/devemux86/cruiser/assets/3484020/c341cb60-90df-4547-963e-ce355bfb0844" width="50%"/>

**Reply by 1maxi2** [reply]
Das ist soweit klar. Aber ich habe z.B. eine fertig geplante Route von Österreich nach Slowenien ohne Autobahn.
Dann komme ich auf die Idee um Zeit zu sparen möchte ich in Ö. die Autobahn benutzen. O.k. ich setze nach der Grenze einen Wegpunkt. Aber was mache ich jetzt das in Ö. Autobahnen benutzt werden in Slo.aber nicht ?
Natürlich kann ich 2 Routen planen. Eine in Ö. und eine in Slo. aber das ist nicht im Sinne des Erfinders. 😏

**Reply by devemux86** [reply]
@1maxi2 

Now you can change the routing options and select "Recalculate section" so that only the selected section is changed.
We will see the BRouter profiles in the near future. :slightly_smiling_face: 

Jetzt können Sie die Routing-Optionen ändern und "Abschnitt-Neuberechnung" wählen, so dass nur der ausgewählte Abschnitt geändert wird.
Wir werden die BRouter-Profile in naher Zukunft sehen. :slightly_smiling_face: 

- https://github.com/devemux86/cruiser/discussions/254

**Reply by 1maxi2** [reply]
O.k. Danke und wie kann ich den Abschnitt auswählen den ich ändern will ?

**Reply by devemux86** [reply]
@1maxi2 

Long press on the route line between the waypoints.
"Recalculate section" only changes the clicked section.

Drücken Sie lange auf die Routenlinie zwischen den Wegpunkten.
"Abschnitt-Neuberechnung" ändert nur den angeklickten Abschnitt.

**Reply by 1maxi2** [reply]
O.k.Danke. Ich dachte ich muss den Abschnitt in der Wegpunktliste auswählen 😏

**Comment by devemux86** [comment]
I updated the BRouter website guide on how to export to GeoJSON or GPX navigation formats:
- https://github.com/devemux86/cruiser/discussions/272

_BRouter website only provides speed limits in GeoJSON format._

**Comment by 0709wiwiwi** [comment]
**Normal routing** create Beeline. 
Edit is predictable. 
Both straight line shaping points can be moved flexibly.
Place shaping points on the street result is  as expected.

**Intelligent routing** create Beeline. 
Edit is quite unstable & confusing.
a. The connection position of the routed with beeline section sometimes false.
   An extra recalcultae all to correct is then needed.
b. Slightly move shaping point 1 result in replacing the shaping point 2.
   Even by this simple example the edit by Intelligent is confusing.

![image](https://github.com/devemux86/cruiser/assets/54167201/824975c6-a12b-4ac5-b495-c97063fde247)
[Straight_normal.gpx.txt](https://github.com/devemux86/cruiser/files/14648681/Straight_normal.gpx.txt)

**Reply by devemux86** [reply]
@0709wiwiwi thanks for the report.
It is simply an error we need to fix. :slightly_smiling_face: 

This straight line case will work properly in the next update.

**Comment by devemux86** [comment]
Cruiser 3.0.42-b1 is available for testing [here](https://www.dropbox.com/sh/itq3e3d0wr0zp3q/AADZ0BYypUf91GhMVKryZzmba):

- https://github.com/devemux86/cruiser/discussions/471

- https://github.com/devemux86/cruiser/discussions/142 improvements
(fix report https://github.com/devemux86/cruiser/discussions/470#discussioncomment-8838405)

If everything is ok, we can have a new release soon.

**Reply by 0709wiwiwi** [reply]
![Intelligent and beeline](https://github.com/devemux86/cruiser/assets/54167201/fc35bbc0-2af7-45ff-9e09-d66ad6f20752)
Image by desktop version.

The same defect happens with Android by Cruiser Intelligent mode. 
But in Android the defect is corrected after a recalculation.

**Reply by devemux86** [reply]
@0709wiwiwi thanks for the report.

Straight lines got complicated, especially now that we want them to work _without calculation_ in all routing services.
I will see how we can improve this case.

**Comment by umpfmpf** [comment]
The icons look pretty!
If you want to revise the optics: It may be nicer if the setting in the BRouter menu from ..% Display in a 6 point display with permanent display and points off-1-2-3-4-5? 
In addition, set the 3 slide switches in the same way: use all on or avoid all of them.

**Reply by devemux86** [reply]
@umpfmpf do you mean the BRouter avoid options to have steps instead of percentages?

**Reply by umpfmpf** [reply]
Yes, like this...
![Screenshot_20240321_091820_Kurviger](https://github.com/devemux86/cruiser/assets/105640513/2050a168-a5b0-4510-9448-468227933067)

**Reply by WolfgangZ42** [reply]
And what is the advantage over the percentage slider? 
"0 20 40 60 80 100" are just as manageable as "off 1 2 3 4 5". Even more understandable in my eyes.

**Reply by devemux86** [reply]
Permanent display will complicate the interface and might confuse users.
Options better display only the value and use slider when we adjust them.

We could use steps or percentage for the label, it's just how the value is presented.
We have discussed this before, but most preferred the percentage as easier to understand.

**Reply by devemux86** [reply]
Note that the above image is a website inside a web view inside a mobile app.
This is not how native mobile components appear.

Also it seems confusing, radio buttons + check boxes simultaneously with 2 colors.
It is not easy to understand what is displayed unless reading some documentation.

**Comment by devemux86** [comment]
Cruiser 3.0.42-b2 is available for testing [here](https://www.dropbox.com/sh/itq3e3d0wr0zp3q/AADZ0BYypUf91GhMVKryZzmba):
- Straight line improvement in intelligent route calculation
(fix report https://github.com/devemux86/cruiser/discussions/470#discussioncomment-8855613)

**Comment by 0709wiwiwi** [comment]
> Straight line improvement in intelligent route calculation.

👍

**Comment by MotoUKRider** [comment]
In "Routing>Options", the title of "Options" when you adjust the avoidance settings may benefit from a tweak, eg: "Options : Motorcycle" so we know what profile we are working on, particularly when getting creative with "Recalculate section".

**Comment by devemux86** [comment]
Cruiser 3.0.42 is on [Google Play](https://play.google.com/store/apps/details?id=gr.talent.cruiser):

- https://github.com/devemux86/cruiser/discussions/142 improvements

- https://github.com/devemux86/cruiser/discussions/471

**Reply by bios4** [reply]
Question @devemux86 regarding the behaviour of Cruiser (current version 3.0.42) during navigation:

1) I have a route loaded and started navigation
2) I long-press the waypoint-panel on the right
3) I select one of the waypoints in my route, the waypoints window closes
4) The distance to this selected waypoint is updated in the waypoint panel, but apart from that nothing happens; the map is still showing my current position (and NOT the waypoint I selected in 3)
5) I short-tap the waypoint panel on the right again
6) The map now centers on the waypoint selected in 3) and switches back to my current position a few seconds later

Is this "two-staged" displaying of the selected waypoint done on purpose, or should the map center on the selected waypoint right after selecting it?

The selected waypoint from 3) and also the shown distance to it is remembered until I again long-press on the waypoint panel and select a different waypoint.

Shouldn't at least the shown distance in the waypoint-panel revert back to the distance to the next waypoint on my route after the position is switched back to my current position? Showing the distance to any other selected waypoint until I update the selection is somehow misleading...

---

Furthermore I'd like to suggest to add an additional setting in Main menu > Settings > Navigation > Display > Route diagram:
- Slider to show/hide it during navigation (available now)
- New slider to show/hide it also in follow mode when a route is loaded, because the long-press on the navigation panel to de-/activate the route diagram is not available in follow mode

**Reply by devemux86** [reply]
> Is this "two-staged" displaying of the selected waypoint done on purpose, or should the map center on the selected waypoint right after selecting it?

Indeed, these are 2 different things, long press for menu or tap for action:
- Long press the waypoint panel at top or right and the waypoint selection screen opens.

- Tap the waypoint panel at top or right (if "Destination **and** waypoint" is selected in settings)
and the map centers (temporarily) on the waypoint.
  - Map centers (temporarily) also when we tap the nearby favorites or POI panels at right.

> Shouldn't at least the shown distance in the waypoint-panel revert back to the distance to the next waypoint on my route after the position is switched back to my current position?

This is a setting done explicitly by the user to keep watching the distance / time to a specific waypoint.

If want to watch only the next waypoint, do not change it and use the road book to see all waypoints:
- https://github.com/devemux86/cruiser/discussions/473 (image 1)

> New slider to show/hide it also in follow mode when a route is loaded, because the long-press on the navigation panel to de-/activate the route diagram is not available in follow mode

Route diagram exists in 2 cases:
- Navigation
- Route planning

Follow mode does not know the route, just follows the GPS location.
The diagram appearing there is from the route planning (turn on / off in routing menu).

Or we can hide the route planning diagram in follow mode.

**Reply by bios4** [reply]
> Indeed, these are 2 different things

So I understand this behaviour is intended and OK...

> If want to watch only the next waypoint, do not change it

How can I "reset" the panel in case I selected a waypoint before?

> The diagram appearing there is from the route planning (turn on / off in routing menu).

Ah, yes this makes sense...
In this case I would vote for "keep everything as is", no need to hide the route diagram explicitly and always in follow mode.

Thanks!
Happy easter!

**Reply by devemux86** [reply]
> How can I "reset" the panel in case I selected a waypoint before?

Waypoints selection screen shows only the waypoints after the user location.

So if want to revert the waypoint panels to display the "next" waypoint,
long press any of them and select the 1st waypoint, it is the "next" one.

**Comment by Cer-o** [comment]
@devemux86  Hace dos días que he sacado la moto del taller, o sea que estoy sin salir de ruta algo más de cuatro meses y por tanto sin manejar "cruiser".
Me da miedo encontrarme una aplicación muy muy diferente a la que conocía hace tan poco tiempo.
Empezaré a toquetearla para familiarizarme con tanta novedad.
Un saludo y gracias por el esfuerzo que haces dando soporte a los usuarios.

**Reply by devemux86** [reply]
@Cer-o Thank you for your kind words. If you have any questions, please write to us and we will help you.

@Cer-o Gracias por sus amables palabras. Si tiene alguna duda, escríbanos y le ayudaremos.

**Comment by Leeb64** [comment]
There seems to be something wrong with the GPS route recording....
The first 300m were recorded correctly. Then a straight line to the first stop (about 80km as the crow flies). Then again 500m correctly to the 2nd stop.
Then nothing more...
The APP is set to always record.

**Reply by devemux86** [reply]
What filter settings do you use in GPS recording?

What location service, Android or Google?
(test both of them)

Also, if have closed screen or the app in background, remove the app from Android battery optimization.
See here: https://dontkillmyapp.com

**Reply by Leeb64** [reply]
Filter: intervall 2s/distance 15m/GPS acuraracy 6m/min Speed 5km/h
Location service : Android
The APP was running the whole time. I used it to navigate!
Energy saving function: excluded !

**Reply by devemux86** [reply]
- If GPS accuracy is low, then records can be excluded (depending on reception and sensor).
Try the default 50 m.

- Also try without the speed filter.

Default values are: time 5s, distance 10m, accuracy 50m, speed off.

**Comment by Leeb64** [comment]
Now it works!

**Reply by devemux86** [reply]
Was it the GPS accuracy filter?

**Reply by Leeb64** [reply]
Probably...
The GPS accuracy is sometimes about 20-30m off.

**Reply by 0709wiwiwi** [reply]
Indeed, this is usually the killjoy. All too often, the accuracy of GPS is estimated far too optimistically. I keep the default setting.  I wonder if it wouldn't be more optimal for Cruiser not to offer this choice option.

**Reply by devemux86** [reply]
We could increase the minimum from 1 to 10 m?

Have you experience with GPS recording apps, what filters do they use?

**Reply by Leeb64** [reply]
I think a lot depends on the accuracy of the GPS sensor.
Maybe you should add a red(unsafe) settings section?

**Reply by Leeb64** [reply]
The choice is actually unnecessary.
What good is an accuracy setting of 1m if it doesn't work properly?

**Reply by 0709wiwiwi** [reply]
> The choice is actually unnecessary.

Indeed plus it results in more simplify.

Similar:  "Skip check accuracy" ( in off route detection):  Useful while no negatives noticed.
https://github.com/devemux86/cruiser/discussions/232#discussioncomment-6476790

**Comment by devemux86** [comment]
Sorry, but I will come back to the topic for a moment. I think I misunderstood that in 3D mode, points closer to us are displayed OVER the points further away. While this is indeed the case for points in one group, this is not the case for different groups. I still think it's the group's position on the group list that determines it - but in reverse order.

3D view:

<img src="https://github.com/devemux86/cruiser/assets/122647782/070fbc42-5360-4244-9476-87db00af68a3" width="33%"/>

Just swap the groups on the list:

<img src="https://github.com/devemux86/cruiser/assets/122647782/2cadd830-039f-4cfd-896f-2092462bbecd" width="33%"/>

_Originally posted by @Ciutnik in https://github.com/devemux86/cruiser/discussions/451#discussioncomment-8774815._

**Reply by devemux86** [reply]
@Ciutnik

In next Cruiser the order of favorites (groups) / routes / tracks in their lists
will also set their z-order while rendering them on the map.

Items higher in the lists will render above items lower in the lists.
